src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -272,11 +272,11 @@ case class VSMSubgen(
     for (size <- Gen.size;
       subsizes <- Gen.partitionSize(5).resize(size / 10);
       vSig <- vSigGen.resize(3);
-      vaSig <- vaSigGen.resize(subsizes(0));
+      vaSig <- vaSigGen.map(t => t.deepOptional()).resize(subsizes(0));
       sSig <- sSigGen.resize(3);
-      saSig <- saSigGen.resize(subsizes(1));
+      saSig <- saSigGen.map(t => t.deepOptional()).resize(subsizes(1));
       globalSig <- globalSigGen.resize(subsizes(2));
-      tSig <- tSigGen.resize(3);
+      tSig <- tSigGen.map(t => t.structOptional()).resize(3);
```

Comment:
this will strip out the required elements in genotype.ad and genotype.px, jfyi

Reply:
Those required are on the representation, but all I should see here is the complex type, Genotype, no?

Reply:
womp. I forgot we were still using TGenotype for VSMSubgen.